navigation/scripts/nav.py

```diff
@@ -0,0 +1,143 @@
+#!/usr/bin/env python
+
+import rospy
+import math
+
+from actionlib import SimpleActionClient
+
+from navigation.srv import NavigateTo, NavigateToResponse
+
+from move_base_msgs.msg import MoveBaseGoal, MoveBaseAction
+from actionlib_msgs.msg import GoalStatusArray
+from geometry_msgs.msg import PoseWithCovarianceStamped, Pose
+
+IDLE = 0
+EXPLORING = 1
+NAVIGATING = 2
+FOLLOWING = 3
+GUIDING = 4
+
+class Navigation:
+
+    def __init__(self):
+
+        # initialise state variables
+        self._reset()
+
+        # services
+        self.robot_pose = Pose()
+        s = rospy.Service('navigate_to', NavigateTo, self.setNavigate)
+
+        # sub
+        rospy.Subscriber('robot_pose', PoseWithCovarianceStamped, self._pose_callback)
+
+        # NavigateTo
+        self.move_base_msg = MoveBaseGoal()
+        self.move_base_msg.target_pose.header.frame_id = 'map'
+
+        self.move_base_client = SimpleActionClient('move_base', MoveBaseAction)
+        self.move_base_client.wait_for_server()
+
+        rospy.loginfo("Ready to navigate")
+
+    def _pose_callback(self, msg):
+        self.robot_pose = msg.pose.pose
+
+    def _reset(self):
+        self.state = IDLE
+        self.goal_pose = None
+        
+        self.move_base_request_sent = False
+
+    ### service functions
+    def setExplore(self):
+        self.state = EXPLORING
+
+    def setNavigate(self, msg):
+        self._reset()
+        self.state = NAVIGATING
+        self.goal_pose = msg.pose
+        return NavigateToResponse(True)
```

Comment:
the service is never going to fail ?

Reply:
can change to empty response, or can leave it for future if we want the nav to be able to reject requests